### PR TITLE
[p] Remove Esy from sandbox scripts

### PR DESCRIPTION
**Pull request by d4hines**
Created in draft mode until we transition the whole team to Nix.

## Depends

- [x] #554

## Problem

If we remove esy from our sandbox script, this yields:
- simpler scripts - no branching based on package manager
- a lot of leverage in packaging the script for clients. This has been really useful to me for parametric deku.

## Solution

Remove esy from the scripts.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -19,13 +19,23 @@
     prometheus-web.inputs.ocaml-overlay.follows = "ocaml-overlays";
 
     nix-npm-buildpackage.url = "github:serokell/nix-npm-buildpackage";
+    tezos.url = "github:marigold-dev/tezos-nix/ulrikstrid/hydra-test";
```
Comment (ulrikstrid) [inline flake.nix:22]:
```suggestion
    tezos.url = "github:marigold-dev/tezos-nix";
```

File: reason-to-ocaml.sh [hunk 2]
```diff
@@ -14,12 +14,12 @@ fi
 
 fix_let_syntax() {
     extensions=("await" "some" "default" "ok" "assert")
-    for extension in ${extensions[@]}; do
+    for extension in "${extensions[@]}"; do
```
Comment (ulrikstrid) [inline reason-to-ocaml.sh:17]: Do we still have reasoan in the codebase or can we drop this/move this somewhere else?

File: sandbox.sh [hunk 3]
```diff
@@ -4,18 +4,7 @@ set -e
```
Comment (ulrikstrid) [inline sandbox.sh:4]: Most of the changes in this file seems to be shellcheck so would be nice to land #554 first

**Comment by ulrikstrid** [comment]
Before this can be considered we must generate docker images from nix, I can open a separate PR for that

**Comment by oteku** [comment]
Nix would be a default dev env = If i don't want to manage my setup I use Nix!

But it must not be the only setup. So OCaml deps must be managed at project level either with Esy or opam switch. Nix must be used only to manage os and global deps.

That say the workflow of Deku doesn't take avantage of esy and opam switch would be more efficient ... but that's another topic

**Comment by EduardoRFS** [comment]
@oteku why? Why not just go with nix for OCaml dependencies?

> That say the workflow of Deku doesn't take avantage of esy and opam switch would be more efficient ... but that's another topic

Which advantage you're talking about? Deku uses all the features provided by esy.

File: reason-to-ocaml.sh (hunk 2 above)
Comment (d4hines) [reply reason-to-ocaml.sh:17]: We have some old prs with reason. After the next push to clean up old pr's, let's remove this.

**Comment by anchpop** [comment]
I'm not able to use Nix currently, I get this error when building:

```
nix develop -c $SHELL
warning: Using saved setting for 'trusted-substituters = https://cache.nixos.org/ https://deku.cachix.org' from ~/.local/share/nix/trusted-settings.json.

[ERROR]: gitstatus failed to initialize.

Please run the following command to install Rosetta:

    /usr/sbin/softwareupdate --install-rosetta

See for details: https://support.apple.com/en-us/HT211861

Once Rosetta is installed, restart your shell.

 ~/coding/marigold/deku | on nix/arm64Compatible #1 ----------------------------------------------------------- impure nix | at 12:32:19
> dune build
File "src/tezos_interop/dune", line 9, characters 0-154:
 9 | (rule
10 |  (deps ./webpack.config.js ./tezos_js_bridge.js)
11 |  (targets ./tezos_js_bridge.bundle.js)
12 |  (mode fallback)
13 |  (action
14 |   (run webpack --env=%{profile})))

<--- Last few GCs --->

[99921:0x140008000]      157 ms: Scavenge 11.2 (16.7) -> 9.3 (17.7) MB, 1.8 / 0.0 ms  (average mu = 1.000, current mu = 1.000) allocation failure
[99921:0x140008000]      210 ms: Scavenge 13.3 (18.4) -> 11.5 (27.9) MB, 1.1 / 0.0 ms  (average mu = 1.000, current mu = 1.000) allocation failure
[99921:0x140008000]      305 ms: Scavenge 19.9 (29.1) -> 15.1 (30.6) MB, 0.9 / 0.0 ms  (average mu = 1.000, current mu = 1.000) allocation failure


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x0001066408d1 <JSObject>
    0: builtin exit frame: new constructor(aka Module)(this=0x000107f90839 <Object map = 0x11243ba89>,0x000107f906e9 <Uint8Array map = 0x106625919>,0x000107f90839 <Object map = 0x11243ba89>)

    1: ConstructFrame [pc: 0x104c5cd18]
    2: StubFrame [pc: 0x104d457c4]
    3: /* anonymous */ [0x107f8a341] [/nix/store/r10mpipi2lnkbd8hpix0bipk0wq4r2x8-sidechain-0.0.0/node_modules/webpack/lib/util/has...

FATAL ERROR: wasm code commit Allocation failed - process out of memory
 1: 0x10446eb48 node::Abort() [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
 2: 0x10446ecc8 node::errors::TryCatchScope::~TryCatchScope() [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
 3: 0x1045976a4 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
 4: 0x104597638 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
 5: 0x104af94c8 v8::internal::wasm::WasmCodeAllocator::AllocateForCodeInRegion(v8::internal::wasm::NativeModule*, unsigned long, v8::base::AddressRegion) [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
 6: 0x104afa414 v8::internal::wasm::NativeModule::CreateEmptyJumpTableInRegion(unsigned int, v8::base::AddressRegion) [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
 7: 0x104af98ac v8::internal::wasm::NativeModule::AddCodeSpace(v8::base::AddressRegion) [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
 8: 0x104afa200 v8::internal::wasm::NativeModule::NativeModule(v8::internal::wasm::WasmEngine*, v8::internal::wasm::WasmFeatures const&, bool, v8::internal::VirtualMemory, std::__1::shared_ptr<v8::internal::wasm::WasmModule const>, std::__1::shared_ptr<v8::internal::Counters>, std::__1::shared_ptr<v8::internal::wasm::NativeModule>*) [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
 9: 0x104afc54c v8::internal::wasm::WasmCodeManager::NewNativeModule(v8::internal::wasm::WasmEngine*, v8::internal::Isolate*, v8::internal::wasm::WasmFeatures const&, unsigned long, bool, std::__1::shared_ptr<v8::internal::wasm::WasmModule const>) [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
10: 0x104b03680 v8::internal::wasm::WasmEngine::NewNativeModule(v8::internal::Isolate*, v8::internal::wasm::WasmFeatures const&, unsigned long, bool, std::__1::shared_ptr<v8::internal::wasm::WasmModule const>) [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
11: 0x104b03588 v8::internal::wasm::WasmEngine::NewNativeModule(v8::internal::Isolate*, v8::internal::wasm::WasmFeatures const&, std::__1::shared_ptr<v8::internal::wasm::WasmModule const>) [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
12: 0x104ad9188 v8::internal::wasm::CompileToNativeModule(v8::internal::Isolate*, v8::internal::wasm::WasmFeatures const&, v8::internal::wasm::ErrorThrower*, std::__1::shared_ptr<v8::internal::wasm::WasmModule const>, v8::internal::wasm::ModuleWireBytes const&, v8::internal::Handle<v8::internal::FixedArray>*) [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
13: 0x104b01420 v8::internal::wasm::WasmEngine::SyncCompile(v8::internal::Isolate*, v8::internal::wasm::WasmFeatures const&, v8::internal::wasm::ErrorThrower*, v8::internal::wasm::ModuleWireBytes const&) [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
14: 0x104b2302c v8::(anonymous namespace)::WebAssemblyModule(v8::FunctionCallbackInfo<v8::Value> const&) [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
15: 0x104600070 v8::internal::FunctionCallbackArguments::Call(v8::internal::CallHandlerInfo) [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
16: 0x1045ff384 v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<true>(v8::internal::Isolate*, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::FunctionTemplateInfo>, v8::internal::Handle<v8::internal::Object>, v8::internal::BuiltinArguments) [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
17: 0x1045fee8c v8::internal::Builtin_Impl_HandleApiCall(v8::internal::BuiltinArguments, v8::internal::Isolate*) [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
18: 0x104cdf68c Builtins_CEntry_Return1_DontSaveFPRegs_ArgvOnStack_BuiltinExit [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
19: 0x104c5cd18 Builtins_JSBuiltinsConstructStub [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
20: 0x104d457c4 Builtins_ConstructHandler [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
21: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
22: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
23: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
24: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
25: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
26: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
27: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
28: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
29: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
30: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
31: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
32: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
33: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
34: 0x104c59fa8 Builtins_ArgumentsAdaptorTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
35: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
36: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
37: 0x104c59fa8 Builtins_ArgumentsAdaptorTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
38: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
39: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
40: 0x104c59fa8 Builtins_ArgumentsAdaptorTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
41: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
42: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
43: 0x104c59fa8 Builtins_ArgumentsAdaptorTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
44: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
45: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
46: 0x104c59fa8 Builtins_ArgumentsAdaptorTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
47: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
48: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
49: 0x104c59fa8 Builtins_ArgumentsAdaptorTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
50: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
51: 0x104c59fa8 Builtins_ArgumentsAdaptorTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
52: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
53: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
54: 0x104c59fa8 Builtins_ArgumentsAdaptorTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
55: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
56: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
57: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
58: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
59: 0x104c59fa8 Builtins_ArgumentsAdaptorTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
60: 0x104c61ae0 Builtins_InterpreterEntryTrampoline [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
61: 0x104c8e484 Builtins_AsyncFunctionAwaitResolveClosure [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
62: 0x104cb2bc0 Builtins_PromiseFulfillReactionJob [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
63: 0x104c80480 Builtins_RunMicrotasks [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
64: 0x104c5ea4c Builtins_JSRunMicrotasksEntry [/nix/store/0lrf3p7ciyinavh088pvy40xxnslzdl0-nodejs-12.22.11/bin/node]
65: 0x140008000
File "_none_", line 1:
Warning 58 [no-cmx-file]: no cmx file was found in path for module Zarith_version, and its interface was not compiled with -opaque

 ~/coding/marigold/deku | on nix/arm64Compatible #1 ------------------------------------------------ took 11s | impure nix | at 12:32:42
>
```

**Pull request closed without merging**